sdk/src/androidTest/java/com/bugsnag/android/SessionTrackingPayloadTest.java

```diff
@@ -93,4 +101,17 @@ public void testMultipleSessionFiles() throws Exception {
         assertEquals(2, sessions.length());
     }
 
+    @Test
+    public void testDefaultOverride() throws Exception {
+        session = new Session("id", new Date(), null, false);
+        Context context = InstrumentationRegistry.getContext();
+        payload = generatePayloadFromSession(context, session);
+        assertFalse(session.isAutoCaptured());
+        session.setAutoCaptured(true);
+        assertTrue(session.isAutoCaptured());
+
+        JSONObject rootNode = streamableToJson(payload);
+        JSONObject sessionNode = rootNode.getJSONArray("sessions").getJSONObject(0);
+        assertFalse(sessionNode.has("user"));
```

Comment:
This seems to imply there should be some other test for `user` contents. Also, shouldn't the fields which exist in the payload be tested?

Reply:
Yep - I believe they're mostly covered in the `testPayloadSerialisation` method, and the serialisation tests for individual classes (e.g. User).